impl/maven-di/src/main/java/org/apache/maven/di/impl/Binding.java

```diff
@@ -128,14 +128,19 @@ public Annotation getScope() {
         return scope;
     }
 
-    public String getDisplayString() {
+    public String getDisplayString() { // fixme unused
         return dependencies.stream().map(Dependency::getDisplayString).collect(joining(", ", "[", "]"));
     }
 
     public Key<?> getOriginalKey() {
         return originalKey;
     }
 
+    @Override
+    public int compareTo(Binding<?> objectBinding) {
+        return Integer.compare(this.priority, objectBinding.priority);
```

Comment:
Reverse the logic, bindings are always sorted in descending order to get the ones with highest priority first.